pkg/reporter/reporter.go

```diff
@@ -0,0 +1,48 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reporter
+
+type Result int
+
+const (
+	Success Result = iota
+	Failure
+	Warning
+)
+
+// Reporter is responsible for reporting back on the progress of the specified preparation.
+type Reporter interface {
+	// Start will report that an operation started.
+	Start(message string, args ...interface{})
+
+	// Success will report that the last operation has succeeded with the specified message.
+	Success(message string, args ...interface{})
+
+	// Failure will report that the last operation has failed with the specified error.
+	Failure(message string, args ...interface{})
+
+	// End ends the Start cli with Success if there are no errors otherwise with a Failure.
+	End(result Result)
+
+	// Warning will report that the last operation has thrown a warning.
+	Warning(message string, args ...interface{})
+
+	// ResultFromMessages returns a Result based on the message queue.
+	ResultFromMessages() Result
```

Comment:
Again, this shouldn't need to be part of the interface - the implementation can internally infer this if needed.

```suggestion
```